Assets/Scripts/PlayerAttack.cs

```diff
@@ -62,17 +69,23 @@ void Update()
 
         ProjectileFXPair GetProjectilePair(WeaponId weaponId)
         {
-            // TODO: replace with mWeaponId indexed array
-            // Return string resource name to instantiate networked projectile
             switch (weaponId)
             {
                 case WeaponId.VETO:
-                    return new ProjectileFXPair("VetoBullet", "VetoFireFX");
+                    return ProjectileFXPair.VETO;
                 case WeaponId.MAGIC_BULLET:
-                    return new ProjectileFXPair("MagicBulletBullet", null);
+                    return ProjectileFXPair.MAGIC_BULLET;
                 default:
-                    return new ProjectileFXPair(null, null);
+                    return ProjectileFXPair.FISTS;
             }
         }
+
+        bool ProjectileSpawnIsntClipping(Vector3 weaponOrigin, Vector3 weaponEmissionPoint)
+        {
+            Vector2 emissionVector = weaponEmissionPoint - weaponOrigin;
+            float barrelDistance = emissionVector.magnitude;
+            var hit = Physics2D.Raycast(weaponOrigin, emissionVector, barrelDistance, mWeaponClippingCheck);

```

Comment:
The weapon origin here is the center of the player and the player's bounding box should never clip past a wall entirely